Assets/Scripts/Enemy/EnemyBehaviour.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/**
+ * Generic EnemyBehaviour. Should be initialized after creation via Initialize(), otherwise is not usable.
+ */
+public class EnemyBehaviour : MonoBehaviour
+{
+    private bool _initialized = false;
+    // Transform of the enemy's target.
+    private Transform _targetTransform;
+    private Quaternion _lookRotation;
+    private Vector3 _direction;
+    private bool _cooldown = false;
+    private float _cooldownDelay = 1f;
+    // _weapon = new Weapon();
+
+    // Method to pathfind from Transform to Transform, returns Vector3, the direction to move in next.
+    private Func<Transform, Transform, Vector3> Pathfind;

```

Comment:
Is this `pathFind` utilizing the mesh pathfinding solution that @ethanakester came across ?

Reply:
A pathfinding algorithm doesn't exist yet. This is only the framework to support a solution, the pathfinding solution should be implemented week 2 IIRC.

Reply:
We should discuss the pathfinding sometime this week as we have an idea of which direction we want to take. (either using Ethan's method or something else)